rmw_opendds_cpp/src/rmw_node.cpp

```diff
@@ -30,47 +30,39 @@ rmw_create_node(
   size_t domain_id,
   bool localhost_only)
 {
-  RCUTILS_CHECK_ARGUMENT_FOR_NULL(context, NULL);
-  RMW_CHECK_TYPE_IDENTIFIERS_MATCH(
-    init context,
-    context->implementation_identifier,
-    opendds_identifier,
-    // TODO(wjwwood): replace this with RMW_RET_INCORRECT_RMW_IMPLEMENTATION when refactored
-    return NULL);
-  return create_node(opendds_identifier, name, namespace_, domain_id);
+  (void)domain_id;
+  (void)localhost_only;
+  RMW_CHECK_FOR_NULL_WITH_MSG(context, "context is null", NULL);
+  if (!check_impl_id(context->implementation_identifier)) {
+    return NULL;
+  }
+  RMW_CHECK_FOR_NULL_WITH_MSG(context->impl, "context->impl is null", NULL);
+  RMW_CHECK_FOR_NULL_WITH_MSG(name, "node name is null", NULL);
+  RMW_CHECK_FOR_NULL_WITH_MSG(namespace_, "node namespace_ is null", NULL);
+  return create_node(*context, name, namespace_);
 }
 
 rmw_ret_t
 rmw_destroy_node(rmw_node_t * node)
 {
-  return destroy_node(opendds_identifier, node);
+  RMW_CHECK_FOR_NULL_WITH_MSG(node, "node is null", RMW_RET_INVALID_ARGUMENT);
+  if (!check_impl_id(node->implementation_identifier)) {
+    return RMW_RET_INCORRECT_RMW_IMPLEMENTATION;
```

Comment:
Shouldn't this print an error here? Maybe it's handled by the caller but since we already have these other null checks printing, its seem well established to print errors at this level.

Reply:
It's done in check_impl_id().